### ISSUE TITLE
standard & improve word choice around inter-vm file transfers

### ISSUE THREAD

**Issue by mfc** [QubesOS/qubes-issues#1040]
in Fedora templates, the Action Menu wording for inter-vm file transfers is a bit ambiguous and confusing:

1. Send to VM
2. Open in Disposable VM
3. Move to VM

"Send to" semantically suggests the original is not kept, which is not true.
 
In Whonix templates the Action Menu is clearer with "Copy to" being used instead of "Send to":

1. Move to Other VM...
2. Copy to Other VM...
3. Open in Disposable VM

Note the different ordering as well. 

I would recommend the implementation of the Whonix template practices in the other templates, both in wording (as they are clearer) and ordering (as isolating the Disposable VM option from the others makes sense).

**Comment by mfc** [comment]
i'm pretty sure this went in the wrong direction @adrelanos.

now whonix, debian, and fedora are all unified in language (yay!) but not ordering, and the language change is the wrong one, to:

1. send to VM
2. move to VM

As originally mentioned "send" is used semantically incorrectly here, it needs to be "copy":

1. copy to VM
2. move to VM

**Comment by adrelanos** [comment]
Michael Carbone:
> i'm pretty sure this went in the wrong direction @adrelanos.

Why me?

**Comment by mfc** [comment]
> Why me?

haha, didn't mean for it to sound scary. the language in Whonix template-derived AppVMs changed from "Copy to Other VM..." to "Send to Other VM" and I figured you are responsible for Whonix template changes?

**Comment by marmarek** [comment]
On Thu, Jan 14, 2016 at 03:01:14AM -0800, Michael Carbone wrote:
> haha, didn't mean for it to sound scary. the language in Whonix template-derived AppVMs changed from "Copy to Other VM..." to "Send to Other VM" and I figured you are responsible for Whonix template changes?

This is just part of qubes packages installed there, the same as in
other templates. Visual differences are because Whonix uses dolphin as
file manager (right?), which is configured separately from Nautilus file
manager (used by default in Fedora templates).

-- 
Best Regards,
Marek Marczykowski-Górecki
Invisible Things Lab
A: Because it messes up the order in which people normally read text.
Q: Why is top-posting such a bad thing?

**Comment by mfc** [comment]
ah thanks for the clarification. I'm highlighting the differences in wording primarily (how "send" should be "copy"), rather than order which is a second issue -- ie the order of the options is different in dolphin and nautilus in Debian-based AppVMs for instance.

**Comment by adrelanos** [comment]
Yes, but I wouldn't change these messages in Whonix only. Especially not
without discussing it in a ticket. These messages are defined by
packages provided by Qubes only.

**Comment by adrelanos** [comment]
Marek Marczykowski-Górecki:
> Visual differences are because Whonix uses dolphin as
> file manager (right?)

Yes.

**Comment by mfc** [comment]
right, so it sounds like it is Dolphin menu language (provided by Qubes) that need to be improved? Whether it is Dolphin in Debian, Whonix, Fedora, etc. That will ensure the user finds the same language (and ideally same order) as in Nautilus.

**Comment by adrelanos** [comment]
Yes.

**Comment by mfc** [comment]
I think this is a pretty easy fix, we just need the Dolphin context menu text string to be the same as the Files/Nautilus language: both "Copy To VM" (rather than "Send To VM").

**Comment by unman** [comment]
@andrewdavidwong Confirmed issue still arises in 3.2 milestone

**Comment by mfc** [comment]
confirming still an issue in 4.0

**Comment by unman** [comment]
@andrewdavidwong @ninavizz 
I've put in PR to fix "Send to" to "Copy to" - so this is consistent across the board.
If there is inconsistency in the *ordering* of items in the menus of different file managers, should this be addressed?
If not, this could be closed once the PRs are  merged.

**Comment by brendanhoar** [comment]
> @andrewdavidwong @ninavizz 
> 
> I've put in PR to fix "Send to" to "Copy to" - so this is consistent across the board.

Don't have a dog in the race (other than consistency), but as I am sure this will come up later: 

Will the option remain under the "Send To" right-click **Explorer** menu in Windows, or will it be moved and/or renamed there as well?

B

**Comment by ninavizz** [comment]
My dog is with @brendanhoar's dog. I really appreciate the flag @unman, as I am very with you in desiring consistency—but on the order of global-project-magnitude there may be bigger fish to fry? Or not? Just the "Copy To" language change feels so big to me, imho. TY for that! 

For once, I do not have a strong opinion on a matter, beyond gratitude for working with other great project contributors. :)

**Comment by deeplow** [comment]
Just adding my 2 cents here. "Copy to other AppVM..." is to hard to parse. I take quite a bit of time to find it due to the length. On the other side of the spectrum is whonix's "Copy to VM", which is perfect for me. But eventually it may turn into "Copy to Qube" due to https://github.com/QubesOS/qubes-issues/issues/1015, which is also fine.

**Comment by andrewdavidwong** [comment]
Maybe the inter-VM actions should have a Qubes icon and/or Qubes-blue background to visually distinguish them from intra-VM actions at a glance.

**Comment by ninavizz** [comment]
@andrewdavidwong I created #5853 to improve the inter-VM action dialogs (text and icons, both) to be clearer; and with @marmarta iterated on a version for a global icon that could work with all actions, that for whatever reason was not in that issue (and I think it was the first thing I ever worked with Marta on). With #4721 this issue feels like a wonderful enhancement of today's Services experience. 

Once #4721 is closer to completion, I'd love to "finish" the UX component of #5853 to move it all into development. Overall, the total "inter-VM experience" across Qubes, could be better. Permissions dialogs, feel like a nice clean chunk to polish. 

![intervm-icon](https://user-images.githubusercontent.com/8262612/119622502-104e4b80-bdbc-11eb-958b-3acd76c1815e.png)

**Comment by deeplow** [comment]
For consistency:

| This layout | could also be applied to the prompt bellow |
|---|---|
|![policy](https://user-images.githubusercontent.com/47065258/119643640-81691f80-be0b-11eb-8960-4f6b1b652475.png) | ![core3-qrexec-prompt](https://user-images.githubusercontent.com/47065258/119644031-f89eb380-be0b-11eb-8a5c-e10a68141f5f.png)

**Comment by ninavizz** [comment]
@deeplow The layout on the left, was only generated as a "concept-model" that would inform the interactions of a form. I'd never want to expose that to users. Sorry if my annotations didn't clearly spell that out.

**Comment by andrewdavidwong** [comment]
> @andrewdavidwong I created #5853 (as an iteration from #5840) to improve the inter-VM action dialogs (text and icons, both) to be clearer [...]

I meant in the context menu itself. By the time you get to the dialog, you've already had to find the correct option in the context menu. Anyway, just a minor suggestion.

**Comment by unman** [comment]
On Wed, May 26, 2021 at 12:58:52PM -0700, Andrew David Wong wrote:
> > @andrewdavidwong I created #5853 (as an iteration from #5840) to improve the inter-VM action dialogs (text and icons, both) to be clearer [...]
> 
> I meant in the context menu itself. By the time you get to the dialog, you've already had to find the correct option in the context menu. Anyway, just a minor suggestion.
> 

Is there anything left in this issue separate from issues covered by
@ninavizz work?
If so, can it be explicitly set out - in words, not linked to an image
please. I'm losing the thread.

**Comment by ninavizz** [comment]
@unman I finally got a chance to look at File Manager menus to more closely examine how users can invoke inter-vm file transfers, yesterday—and, oyy, yes, the menus totally need some love. Is it possible to create "sections" in menus (as [this html example](https://www.patternfly.org/v4/components/simple-list/react/grouped-list/) from PatternFly suggests)?

If so, I feel the best approach would be to create a section for intra-qube actions, and a section for inter-qube actions. 

If it is possible to both create sections, and to create unique styling for the text of section group names (per the HTML patternfly example) let me know and I'll gladly create a wireframe for the Qubes file menus. @unman are visual examples—PDFs, PNGs, etc—not parsable by you... and would HTML help you, more?

**Comment by ninavizz** [comment]
@unman This is a mockup I put together for another project. Attempted an all-text version, below the image in this comment. Happy to provide HTML/CSS, if this might be something you'd want to implement across file managers (assuming others and @marmarek are also into this).

**Summation of proposal:** grouping all app-unique/specific/intra-qube actions above, and all inter-qube and Qubes OS facilitated things, below in a named group. For Gnome File Manager, XFCE/Thunar, and Dolphin. 

<img width="702" alt="Screen Shot 2021-05-31 at 7 47 31 PM" src="https://user-images.githubusercontent.com/8262612/120259634-30ad5880-c249-11eb-9ba7-60eb4d321d0b.png">


> Open with Text Editor		Ctrl+O
> Open with…  			
> ————————————————
> Cut				       Ctrl+X
> Copy				       Ctrl+C
> Move to…
> Copy to…
> Move to Trash 			Delete
> Rename				F2
> Properties			        Ctrl+1
> 
> QUBES UTILITIES ———————
> 
> Compress…
> Convert to Trusted PDF
> Convert to Trusted IMG
> ————————————————
> Copy to another qube
> Move to another qube
> ————————————————
> Edit in Disposable qube
> View in Disposable qube

**Comment by andrewdavidwong** [comment]
I think this separation is very helpful. Just a small note: ￼"Compress…" should be on top￼.

**Comment by ninavizz** [comment]
@andrewdavidwong There's tofu between the "p" and the "." at the end of your sentence—so I'm not 100% sure what you wrote/meant, but edited the above assuming you meant just to the top group (vs the top-most list item).

**Comment by ninavizz** [comment]
Question: If I'm already in a disposable qube, it feels a bit silly to have the "Edit in disposable qube" and "View in disposable qube" options present in the menu. Would it be possible to eliminate or disable those on instances in a disposable qube, or would that be an epic amount of work?

**Comment by brendanhoar** [comment]
> Question: If I'm already in a disposable qube, it feels a bit silly to have the "Edit in disposable qube" and "View in disposable qube" options present in the menu. Would it be possible to eliminate or disable those on instances in a disposable qube, or would that be an epic amount of work?

There is a subset of the user base that may have an initial disposable qube/vm with network access, which then points to a non-networked disposable qubes/vm for handling potentially malicious files, such that they cannot "call home" and report content/network info of disposable qube/vm the file was downloaded into.

**Comment by andrewdavidwong** [comment]
> Question: If I'm already in a disposable qube, it feels a bit silly to have the "Edit in disposable qube" and "View in disposable qube" options present in the menu. Would it be possible to eliminate or disable those on instances in a disposable qube, or would that be an epic amount of work?

It may feel silly to some users, but it is still legitimate functionality with legitimate use cases, so I would recommend against disabling or removing.

**Comment by deeplow** [comment]
> ————————————————
> Copy to another qube
> Move to another qube
> ————————————————

The text looks a bit too long than it needs and it's a bit hard to parse. I would suggest instead something like:

> ————————————————
> Copy to qube
> Move to qube
> ————————————————

On the disposable ones there is probably no way around it

**Comment by ninavizz** [comment]
@deeplow lol, reducing the text length of "Open in another application" had me so pleased I entirely overlooked the `Copy to another qube` text also needing a trim. TY, for that catch! Yeah, the disposable ones there's totally not another way around, but they look clearly more distinct from the ones just above, with the decreased character counts—so I'm happy. TY! :)

**Comment by jmynes** [comment]
Of note, per the conversation around `Send to` verbiage
All of the options in Windows qubes are under a menu literally labeled `Send to`

**Comment by mfc** [comment]
just to update this, as we have gotten a lot closer. for debian & fedora templates they are now the same:
* Copy to other qube
* Move to other qube
* Open in other qube
* Edit in disposable qube
* View in disposable qube

however in whonix-workstation the order is alphabetical:
* Copy to other qube
* Edit in disposable qube 
* Move to other qube
* Open in other qube
* View in disposable qube

as a note they are using PCManFM-Qt File Manager, rather than thunar.

i feel like this issue can be closed as the language is now standardized & improved. and we can make minor issue in whonix to fix the order on their side @adrelanos.
